libenv/sysinfo.c

```diff
@@ -2311,6 +2317,31 @@ static int Linux_Mandriva_Version_Real(EvalContext *ctx, char *filename, char *r
 
 /******************************************************************/
 
+static int Linux_Amazon_Version(EvalContext *ctx)
+{
+    char buffer[CF_BUFSIZE];
+
+    // Amazon Linux AMI release 2016.09
+
+    if (ReadLine("/etc/system-release", buffer, sizeof(buffer)))
+    {
+       if (strstr(buffer, "Amazon"))
+       {
+           char version[CF_MAXVARSIZE], class[CF_MAXVARSIZE];
+           EvalContextClassPutHard(ctx, "amazon", "inventory,attribute_name=none,source=agent");
```

Comment:
@skreuzer I agree with @nickanderson but let's use `derived-from-file=...` since you're right that `derived-from` is for variables. The tag usage has to start somewhere, so rather than a big pull request, I think this is a good place.